packages/flutter/lib/src/material/about.dart

```diff
@@ -472,6 +482,294 @@ class LicensePage extends StatefulWidget {
 }
 
 class _LicensePageState extends State<LicensePage> {
+  final ValueNotifier<int> selectedId = ValueNotifier<int>(null);
+
+  final Future<_LicenseData> licenses =
+      LicenseRegistry.licenses.fold<_LicenseData>(
+    _LicenseData(),
+    (_LicenseData previous, LicenseEntry license) {
+      return previous..addLicense(license);
+    },
+  );
+
+  @override
+  Widget build(BuildContext context) {
+    return _MasterDetailFlow(
+      detailPageFABlessGutterWidth: _getGutterSize(context),
+      title: Text(MaterialLocalizations.of(context).licensesPageTitle),
+      masterViewBuilder: _buildPackagesView,
+      detailPageBuilder: _buildPackageLicensePage,
+    );
+  }
+
+  Widget _buildPackagesView(final BuildContext _, final bool isLateral) {
```

Comment:
here and in other places where you have build methods, might be better to factor those out into their own widgets